scalameta/semantic/shared/src/main/scala/scala/meta/semantic/Database.scala

```diff
@@ -0,0 +1,187 @@
+package scala.meta
+package semantic
+
+import java.io._
+import java.net.URI
+import java.util.zip._
+import scala.{Seq => _}
+import scala.collection.immutable.Seq
+import scala.collection.mutable
+import scala.compat.Platform.EOL
+import scala.util.Try
+import org.scalameta.data._
+import scala.meta.io._
+import scala.meta.internal.io.PlatformIO
+import scala.meta.internal.semantic.codecs._
+import scala.meta.internal.semantic.{proto => p}
+
+@data class Database(sources: Seq[AttributedSource]) {
+  locally {
+    val paths = sources.map(_.path.toString).toSet
+    if (sources.size != paths.size) {
+      val occs = mutable.Map[String, Int]().withDefault(_ => 0)
+      sources.foreach(source => occs(source.path.toString) += 1)
+      val duplicates = occs.filter(_._2 > 1).map(_._1)
+      sys.error("duplicate paths for " + occs.mkString(", "))
+    }
+  }
+
+  @deprecated("Use names instead", "1.8.0") def symbols: Map[Anchor, Symbol] = names
+  def names: Map[Anchor, Symbol] = sources.foldLeft(Map[Anchor, Symbol]())(_ ++ _.names)
+  def messages: Seq[Message] = sources.flatMap(_.messages)
+  def denotations: Map[Symbol, Denotation] = sources.foldLeft(Map[Symbol, Denotation]())(_ ++ _.denotations)
+  def sugars: Map[Anchor, String] = sources.foldLeft(Map[Anchor, String]())(_ ++ _.sugars)
+
+  override def toString: String = sources.mkString(EOL + EOL)
+  def toBinary: Array[Byte] = this.toProto[p.Database].toByteArray
```

Comment:
Following discussions in https://github.com/scalameta/scalameta/issues/777 

I believe we need to pass a `sourceRoot: AbsolutePath` here in order to relativize any `AbsolutePath` inside this Database.